reporter/internal/pdata/generate.go

```diff
@@ -177,27 +190,38 @@ func (p *Pdata) setProfile(
 			default:
 				// Store interpreted frame information as a Line message:
 				line := loc.Line().AppendEmpty()
-
 				if si, exists := p.Frames.GetAndRefresh(
 					libpf.NewFrameID(traceInfo.Files[i], traceInfo.Linenos[i]),
 					FramesCacheLifetime); exists {
 					line.SetLine(int64(si.LineNumber))
-
-					line.SetFunctionIndex(createFunctionEntry(funcMap,
-						si.FunctionName, si.FilePath))
+					fi := samples.FuncInfo{
+						Name:     si.FunctionName,
+						FileName: si.FilePath,
```

Comment:
You could put the strings into the string table here, and make `FuncInfo` to just hold the two string table indices.
And since `FuncInfo` is only used in `generate.go`, no longer expose the type.